source/change-streams/change-streams.rst

```diff
@@ -481,16 +600,20 @@ Imagine a scenario in which a user wants to process each change to a collection
 
 .. code:: python
 
-  resumeToken = None
   localChange = getChangeFromLocalStorage()
+  resumeToken = getResumeTokenFromLocalStorage()
+
   if localChange:
     processChange(localChange)
-    resumeToken = localChange['_id']
 
   try:
-      for change in db.collection.watch([...], resumeAfter=resumeToken):
-          persistToLocalStorage(change)
-          processChange(change)
+      change_stream = db.collection.watch([...], resumeAfter=resumeToken)
+      while True:
+          change = change_stream.try_next()
+          persistResumeTokenToLocalStorage(change_stream.get_resume_token())
```

Comment:
I think this needs to be called twice, before and after iterating the changes. Use case which I believe the current code does not cover: there is a single change made 1 hour ago, the document's resume token in this case should be 1 hour old but post-batch resume token could be minutes old.

Reply:
Perhaps I'm misunderstanding. In this example I'm using `try_next`. That should not block until a document is iterated.